lib/analyzer/image-inspector.ts

```diff
@@ -69,6 +77,14 @@ async function getImageArchive(
   const saveLocation: string = path.join(destination.name, "image.tar");
 
   try {
+    const inspectResult = await getInspectResult(docker, targetImage);
+    if (
+      platform !== undefined &&
+      !isLocalImageSameArchitecture(platform, inspectResult.Architecture)
+    ) {
+      throw new Error("Pulling an image with updated architecture required");
```

Comment:
throwing here is not ideal, but might be good enough for now

Reply:
Yes, 100%! The `getImageArchive()` function is built based on throwing errors 😬 so for me not to throw an error, it would need a bit of refactoring, which I think would be great but maybe not a part of this task?

Reply:
Maybe when adding error handling it will be a good time :-)